airflow/jobs.py

```diff
@@ -2540,9 +2542,32 @@ def _execute(self, session=None):
                         self.dag_id
                     )
                     time.sleep(self.delay_on_limit_secs)
+
+        except (KeyboardInterrupt, SystemExit):
+            for dag_run in self.dag_runs:
+                dag_run.refresh_from_db(session)
+                make_transient(dag_run)
+
+                # Check all the tasks which are not in success state.
+                check_state = State.unfinished() + State.finished()
+                check_state.remove(State.SUCCESS)
+
+                unfinished_tasks = dag_run.get_task_instances(
+                    state=check_state,
+                    session=session
+                )
+                if unfinished_tasks:
+                    # if there are unfinished tasks and ctrl^c
+                    # set dag run state to failed
+                    for task in unfinished_tasks:
+                        task.set_state(State.FAILED, session)
```

Comment:
Does this actually kill any running tasks?

Reply:
my understanding is that we mostly run backfill with local scheduler which spin off couples of worker multiprocess to run the task. Once we set the task state to failed, the task will get rerun next time(using clear first, or with --rerun_failed_task). And if we ctrol^c the main backfill process, it will kill the worker multiprocesses as well which won't leave any orphan task still keep running.